packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -278,17 +272,30 @@ const CompactInteractiveList = (props) => {
   // map rows differently depending on vertical or horizontal orientation
   const mapRows = () => {
     const placeholdersNumber = isResponsive ? (numberOfRows * numberOfColumns) - rows.length : 0;
-    const result = [...rows];
-    // add placeholder rows to the end.
+    let result = [];
+    if (flowHorizontally) {
+      result = rows.map((row, index) => ({ ...row, rowIndex: index }));
+    } else {
+      for (let i = 0; i < numberOfRows; i += 1) {
+        let x = numberOfColumns - placeholdersNumber;
+        for (let j = i; j < rows.length; j += numberOfRows - (x >= 0 ? 0 : 1)) {
+          if (result.length < rows.length) {
+            result.push({ ...rows[j], rowIndex: j });
+            if (x >= 0) { x -= 1; }
+          }
+        }
+      }
+    }
+    // add placeholder rows
     for (let i = rows.length; i < rows.length + placeholdersNumber; i += 1) {
       result.push({ id: `placeholder-row-${i - rows.length + 1}` });
     }
     return result;
   };
 
   const mappedRows = mapRows();
-  const checkIfRowIsLeftMost = (index) => (flowHorizontally ? index % numberOfColumns === 0 : index < rowsPerColumn);
-  const checkIfRowIsTopMost = (index) => (flowHorizontally ? index < numberOfColumns : index % rowsPerColumn === 0);
+  const isLeftmost = (index) => (index % numberOfColumns === 0);
+  const isTopMost = (index) => (index < numberOfColumns);
```

Comment:
nit: can this also be `isTopmost` so the casing matches `isLeftmost`?

Reply:
Fixed in https://github.com/cerner/terra-framework/pull/2047/commits/964a57aa0e629cd0e55e49a9faddf2299e2d2982